test/api.js

```diff
@@ -420,6 +419,157 @@ test('GET /', function(t) {
+      });
+    });
+
+    t.test('missing `AuthSession` value', function(t) {
+      request({
+        method: 'DELETE',
+        uri: uri,
+        json: true,
+        headers: {
+          cookie: 'foobar'
+        }
+      }, function(err, response, body) {
+        t.equal(response.statusCode, 400);
+        t.equal(body.error, 'Bad Request');
+        t.equal(body.message, 'Invalid cookie header');
+        t.end();
+      });
+    });
+
+    t.test('cookie with malformed `AuthSession` value', function(t) {
+      request({
+        method: 'DELETE',
+        uri: uri,
+        json: true,
+        headers: {
+          cookie: 'AuthSession=foobar'
+        }
+      }, function(err, response, body) {
+        t.equal(response.statusCode, 400);
+        t.equal(body.error, 'Bad Request');
+        t.equal(
+          body.message,
+          'Malformed AuthSession cookie. Please clear your cookies.');
+        t.end();
+      });
+    });
+
+    t.test('cookie with invalid `AuthSession` value', function(t) {
+      request({
+        method: 'DELETE',
+        uri: uri,
+        json: true,
+        headers: {
+          cookie: 'AuthSession=Zm9vYmF0b3JAZXhhbXBsZS5jb206NTVFMzJDODc6lYf4JMY9PZaz5tGyRNi8wmlrcwc'
+        }
+      }, function(err, response, body) {
+        t.equal(response.statusCode, 401);
+        t.equal(body.error, 'Unauthorized');
+        t.end();
+      });
+    });
+
+    t.end
+  });
+
+  t.test('deletes the session', function(t) {
+    request({
+      method: 'POST',
+      uri: uri,
+      json: true,
+      body: {
+        email: email
+      }
+    }, function(err, response, body) {
+      var link;
+
+      t.equal(response.statusCode, 200);
+      t.ok(body.ok);
+
+      link = emailBody.match(/([^ ]+)$/)[1];
+
+      request({
+        method: 'GET',
+        uri: link,
+        json: true,
+        followRedirect: false
+      }, function(err, response, body) {
+        var cookie = response.headers['set-cookie'][0].split(';')[0];
+
+        var sessionRequestOptions = {
+          method: 'GET',
+          uri: config.couchDbSessionUrl,
+          json: true,
+          headers: {
+            cookie: cookie
+          }
+        };
+
+        request(sessionRequestOptions, function(err, response, body) {
+          t.equal(response.statusCode, 200);
+          t.ok(body.ok);
+          t.equal(body.userCtx.name, email);
+
+        request({
+          method: 'DELETE',
+          uri: uri,
+          json: true,
+          headers: {
+            cookie: cookie
+          }
+        }, function(err, response, body) {
+          t.equal(response.statusCode, 200);
+          t.ok(body.ok);
+
+            request(sessionRequestOptions, function(err, response, body) {
+              t.equal(response.statusCode, 200);
+              t.ok(body.ok);
+              t.equal(body.userCtx.name, null);
+              t.end();
+            });
+          });
+        });
+      });
+    });
+  });
+
+  t.end();
```

Comment:
it is not about the tests.

the `t.end()` call and `t.end` -- without calling are superfluous in the
both cases

On Sun, Aug 30, 2015 at 9:49 PM, Klaus Trainer <notifications@github.com>
wrote:

> In test/api.js
> <https://github.com/KlausTrainer/couch_email_auth/pull/48#discussion_r38275166>
> :
>
> > +          t.equal(response.statusCode, 200);
> > +          t.ok(body.ok);
> > +
> > +            request(sessionRequestOptions, function(err, response, body) {
> > +              t.equal(response.statusCode, 200);
> > +              t.ok(body.ok);
> > +              t.equal(body.userCtx.name, null);
> > +              t.end();
> > +            });
> > +          });
> > +        });
> > +      });
> > +    });
> > +  });
> > +
> > +  t.end();
>
> There's no other test that proofs that a session is actually invalidated
> by that API call, and I want to be sure that this works.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/KlausTrainer/couch_email_auth/pull/48/files#r38275166>
> .
>